ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/PDEPlugin.java

```diff
@@ -214,6 +216,18 @@ public void start(BundleContext context) throws Exception {
 		LogFilesManager.addLogFileProvider(fLogFileProvider);
 
 		TargetStatus.initializeTargetStatus();
+		registerProgressIcon();
+	}
+
+	protected void registerProgressIcon() {
+		if (!PlatformUI.isWorkbenchRunning()) {
+			return;
+		}
+		IProgressService service = PlatformUI.getWorkbench().getProgressService();
+		if (service == null) {
+			return;
+		}
+		service.registerIconForFamily(PDEPluginImages.DESC_TARGET_DEFINITION, LoadTargetDefinitionJob.getFamily());
```

Comment:
Can't this be simply done in a static initializer in `LoadTargetDefinitionJob`?
This would also solve the problem where to place the job-family constant.

Reply:
The job lives in pde.core. The registration for the icon happens with the workbench progress service, which looks like a UI dependent class to me. I'm not sure if that service can be retrieved in the plain pde.core bundle without disturbing the workbench lifecycle. In egit we therefore even put that code into a service reacting to the application has started lifecycle.
Do you know better?

Reply:
I think given the current structure, this is fine, an alternative might be to use an addon in the e4xmi but that really seems a bit overhead here...

Reply:
> Do you know better?

You are right and no, I don't know better in this case.